source/common/conn_pool/conn_pool_base.h

```diff
@@ -211,6 +215,10 @@ class ConnPoolImplBase : protected Logger::Loggable<Logger::Id::pool> {
 
   bool hasActiveStreams() const { return num_active_streams_ > 0; }
 
+  void incrConnectingStreamCapacity(int32_t delta) {
+    state_.incrConnectingStreamCapacity(delta);
+    connecting_stream_capacity_ += delta;
```

Comment:
Can we have ASSERTs here that this doesn't underflow? Same below? Should the param here be unsigned? Seems off to have incr/decr and also signed.